osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs

```diff
@@ -198,23 +202,24 @@ private void updateButtonColour()
 
                 case MultiplayerUserState.Spectating:
                 case MultiplayerUserState.Ready:
-                    if (room?.Host?.Equals(localUser) == true && !room.ActiveCountdowns.Any(c => c is MatchStartCountdown))
+                    if (multiplayerClient.IsHost && !room.ActiveCountdowns.Any(c => c is MatchStartCountdown))
                         setGreen();
                     else
                         setYellow();
 
                     break;
-            }
 
-            void setYellow()
-            {
-                BackgroundColour = colours.YellowDark;
+                // Show the abort button for the host as long as gameplay is in progress.
+                case MultiplayerUserState when multiplayerClient.IsHost && room.State != MultiplayerRoomState.Open:
```

Comment:
I'm not sure why this has gone unchecked for this long but I'm at the stage where the duality of managing this button bothers me. The visual appearance of the button is managed inside of it, but the actual function it performs changes depending on the decisions of `MatchStartControl`.

This may be alleviated if the abort button isn't comingled with all of it, but here again, I'm not sure I have a very good ready-made solution to offer.

Reply:
Maybe splitting the button into multiple implementations could be tried afterwards. Without having done anything, I tend to favour keeping the conditionals as simple as possible as even if the buttons are split there's still going to have to be some kind of orchestrator (if that's the correct term for it) tying them all together in a hopefully more understandable way.

One other alternative I can imagine is a priority system, where the conditions for each state are split into simple enum flags (e.g. `Abort = IsHost && MatchStarted`) and the priority system picks the most relevant flag in order.